projects/kit/components/input-date-multi/input-date-multi.component.ts

```diff
@@ -0,0 +1,339 @@
+    implements TuiWithOptionalMinMax<TuiDay>, TuiFocusableElementAccessor
+{
+    @ViewChild(TuiPrimitiveTextfieldComponent)
+    private readonly textfield?: TuiPrimitiveTextfieldComponent;
+
+    @ViewChild(TuiInputTagComponent)
+    private readonly inputTag?: TuiInputTagComponent;
+
+    private month: TuiMonth | null = null;
+
+    @Input()
+    min: TuiDay | null = this.options.min;
+
+    @Input()
+    max: TuiDay | null = this.options.max;
+
+    @Input()
+    disabledItemHandler: TuiBooleanHandler<TuiDay> = ALWAYS_FALSE_HANDLER;
+
+    @Input()
+    markerHandler: TuiMarkerHandler = TUI_DEFAULT_MARKER_HANDLER;
+
+    @Input()
+    defaultActiveYearMonth = TuiMonth.currentLocal();
+
+    @Input()
+    expandable = false;
+
+    @Input()
+    inputHidden = false;
+
+    @Input()
+    @HostBinding('class._editable')
+    editable = true;
+
+    @Input()
+    search: string | null = '';
+
+    @Input()
+    placeholder = '';
+
+    @Input()
+    rows = Infinity;
+
+    maskitoOptions: MaskitoOptions = maskitoDateOptionsGenerator({
+        mode: 'dd/mm/yyyy',
+        separator: '.',
+        min: this.min?.toLocalNativeDate(),
+        max: this.max?.toLocalNativeDate(),
+    });
+
+    internalValue: readonly TuiDay[] = [];
+
+    open = false;
+
+    readonly filler$: Observable<string> = this.dateTexts$.pipe(
+        map(dateTexts =>
+            changeDateSeparator(dateTexts[this.dateFormat], this.dateSeparator),
+        ),
+    );
+
+    constructor(
+        @Optional()
+        @Self()
+        @Inject(NgControl)
+        control: NgControl | null,
+        @Inject(ChangeDetectorRef) cdr: ChangeDetectorRef,
+        @Inject(Injector) private readonly injector: Injector,
+        @Inject(TUI_IS_MOBILE) readonly isMobile: boolean,
+        @Inject(TuiDialogService) private readonly dialogs: TuiDialogService,
+        @Optional()
+        @Inject(TUI_MOBILE_CALENDAR)
+        private readonly mobileCalendar: Type<Record<string, any>> | null,
+        @Inject(TUI_DATE_FORMAT) readonly dateFormat: TuiDateMode,
+        @Inject(TUI_DATE_SEPARATOR) readonly dateSeparator: string,
+        @Inject(TUI_DATE_TEXTS)
+        readonly dateTexts$: Observable<Record<TuiDateMode, string>>,
+        @Optional()
+        @Inject(TUI_DATE_VALUE_TRANSFORMER)
+        override readonly valueTransformer: AbstractTuiValueTransformer<
+            readonly TuiDay[]
+        > | null,
+        @Inject(TUI_INPUT_DATE_OPTIONS) private readonly options: TuiInputDateOptions,
+        @Inject(TUI_TEXTFIELD_SIZE)
+        private readonly textfieldSize: TuiTextfieldSizeDirective,
+        @Inject(TUI_DONE_WORD) readonly doneWord$: Observable<string>,
+    ) {
+        super(control, cdr, valueTransformer);
+    }
+
+    @Input()
+    tagValidator: TuiBooleanHandler<string> = (tag: TuiDay | string) => {
+        const {year, month, day} = tuiIsString(tag)
+            ? TuiDay.parseRawDateString(tag)
+            : tag;
+
+        return (
```

Comment:
I think this can be simplified. At the very least you do not need to convert to native dates.

Reply:
fixed